src/codelta_backend/src/lib.rs

```diff
@@ -63,72 +57,90 @@ async fn check_status(canister_id: Principal) -> Result<CanisterStatusResponse,
     Ok(status_response)
 }
 
-async fn _check_balance() -> Result<Tokens, String> {
+async fn _check_balance(sub_account: Subaccount) -> Result<Tokens, String> {
     // Instantiate client for the ICP Ledger (ryjl3-tyaaa-aaaaa-aaaba-cai)
-    let client = ICRC1Client { runtime: CdkRuntime, ledger_canister_id: Principal::from_text(LEDGER_CANISTER_ID).unwrap() };
-    // Default canister account (no subaccount)
-    let canister_account = Account { owner: ic_cdk::id(), subaccount: None };
+    let ledger_canister_id = Principal::from_text(LEDGER_CANISTER_ID).unwrap();
+    // Canister account args
+    let account_balance_args = AccountBalanceArgs {
+        account: AccountIdentifier::new(&ic_cdk::id(), &sub_account),
+    };
     // Get the canister's current ICP balance
-    let canister_balance_e8: Nat = match client.balance_of(canister_account.clone()).await {
-        Ok(balance) => balance,
+    match account_balance(ledger_canister_id, account_balance_args).await {
+        Ok(balance) => Ok(balance),
         Err((_code, err_msg)) => {
             let msg = format!("Failed to get canister balance: {}", err_msg);
             ic_cdk::println!("{}", msg);
             return Err(msg);
         }
-    };
-    Ok(Tokens::from_e8s(canister_balance_e8.0.try_into().unwrap()))
+    }
 }
 
 #[update(guard = "only_members_guard")]
-async fn check_balance() -> Result<String, String> {
-    Ok(_check_balance().await.unwrap().to_string())
+async fn check_balance(topic: Topic) -> Result<String, String> {
+    let team = get_team_by_topic(topic);
+    let team_sub_account = Subaccount(team.sub_account.clone());
+    Ok(_check_balance(team_sub_account).await.unwrap().to_string())
 }
 
+#[update(guard = "only_members_guard")]
+async fn check_subaccount_hex(topic: Topic) -> Result<String, String> {
+    let team = get_team_by_topic(topic);
+    let team_sub_account = Subaccount(team.sub_account.clone());
+    let team_subaccount_hex = AccountIdentifier::new(&ic_cdk::id(), &team_sub_account).to_hex();
+    Ok(team_subaccount_hex)
+}
+
+/// ## Arguments
+/// * `topic` - Proposal topic which is unique per team
+/// * `account_ids` - Optional (useful in case you want to exclude some team accounts - they might have been on vacation)
```

Comment:
Nice touch! I was thinking about this sort of scenario just now. Glad to see you've accounted for it